dali/kernels/alloc.cc

```diff
@@ -27,8 +27,8 @@ struct Allocator;
 
 template <>
 struct Allocator<AllocType::Host> {
-  static void Deallocate(void *ptr, int device) noexcept {
-    (void)device;
+  static void Deallocate(void *ptr, std::shared_ptr<CUContext> &device_context) noexcept {
```

Comment:
This is a bad idea. CUContext requires rework and should be passed here by plain const reference, without extra shared_ptr layer - _if_ we even need that context here, which I don't think we do.